### ISSUE TITLE
Installation of service does not work as advertised

### ISSUE THREAD

**Issue by keithy** [cockpit-project/cockpit-container#20]
This does work (testing on FCOS 20200108.2.0)

```
sudo podman container runlabel INSTALL cockpit/ws cockpit/ws
```
I suspect that the dockerfile needs to be amended like so
```
LABEL INSTALL="/usr/bin/docker run --rm --privileged -v /:/host IMAGE /container/atomic-install IMAGE"
```

**Comment by wdouglascampbell** [comment]
Actually this doesn't work.  It is not creating a properly named container.  So starting the service works but stopping it fails since the container is not being named cockpit-ws.

It seems like the --name parameter of podman doesn't actually work on Fedora CoreOS.  Podman version 1.7.0.

So, unfortunately this seems to be an issue with podman which I don't know where to report that.

For reference here is the cockpit.service file generated if we use keithy's test above.

[Unit]
Description=cockpit/ws podman container

[Service]
Type=oneshot
ExecStartPre=-/usr/bin/podman rm -f cockpit-ws
ExecStart=/usr/bin/podman container runlabel --name cockpit-ws RUN cockpit/ws
ExecStop=/usr/bin/podman rm -f cockpit-ws
RemainAfterExit=yes
KillMode=none

[Install]
WantedBy=multi-user.target

**Comment by martinpitt** [comment]
This was fixed half a year ago: https://github.com/cockpit-project/cockpit/commit/52a172c620bb1eacf5

**Comment by wdouglascampbell** [comment]
> This was fixed half a year ago: [cockpit-project/cockpit@52a172c](https://github.com/cockpit-project/cockpit/commit/52a172c620bb1eacf5)

Yes BUT someone apparently UNFIXED it. If you look at the current version this change is no longer there!

**Comment by martinpitt** [comment]
@wdouglascampbell : I re-tested this now, following the install instructions on https://hub.docker.com/r/cockpit/ws/ . This works  just fine:
```
# podman container runlabel INSTALL cockpit/ws
# systemctl start cockpit
# podman ps -a
CONTAINER ID  IMAGE                        COMMAND               CREATED        STATUS            PORTS  NAMES
65e74643dc27  docker.io/cockpit/ws:latest  /container/atomic...  3 seconds ago  Up 3 seconds ago         cockpit-ws
```
Please be more specific what you are trying to do and how it fails.

**Comment by wdouglascampbell** [comment]
I also just retested this on a fresh FCOS install and it still doesn't work but for different reasons I guess so don't know if the original issue was ever resolved.

```
[root@coreos container-build-area]# podman container runlabel INSTALL cockpit/ws
Trying to pull registry.fedoraproject.org/cockpit/ws...
Error: unable to find image: unable to pull cockpit/ws: Error reading manifest latest in registry.fedoraproject.org/cockpit/ws: manifest unknown: manifest unknown
```

It appears that if I retrieve and build myself then it does work to when running ```podman container runlabel INSTALL cockpit/ws``` but there is no cockpit service file which then results in

```
[root@coreos cockpit-container]# systemctl start cockpit
Failed to start cockpit.service: Unit cockpit.service not found.
```

The atomic-install script is testing the variable $container to see if it contains "podman".  It is empty as far as I can see and I can see no reference to this variable being set anywhere.

**Comment by wdouglascampbell** [comment]
In fact the reason the atomic-install script doesn't work to install the cockpit.service file is that it is checking $container variable for the string "podman" but on FCOS the value is "oci".  https://github.com/containers/podman/issues/3586#issuecomment-598232372 seems to indicate that instead $container should just be checked whether it is set.

Changing the current line in atomic-install from:

```
if [ "${container:-}" = podman ] && [ -n "$IMAGE" ] && [ ! -e /host/etc/systemd/system/cockpit.service ]; then
```

to:

```
if [ ! -z ${container+x} ] && [ -n "$IMAGE" ] && [ ! -e /host/etc/systemd/system/cockpit.service ]; then
```

causes the cockpit.service file to be created when I test it on a fresh install of FCOS.

**Comment by martinpitt** [comment]
I was also testing this on current FCOS, so the environment should be comparable. **However**, I had the cockpit/ws container image already  downloaded. cockpit/ws is indeed not present in the  Fedora registry, and it seems `podman pull` has stopped looking into the docker.io registry by default (!?!) It does work with

    podman container runlabel INSTALL docker.io/cockpit/ws

I updated the [docker hub page](https://hub.docker.com/r/cockpit/ws) accordingly, and sent https://github.com/cockpit-project/cockpit/pull/14380 to fix the README upstream.

I can't reproduce `$container` being `oci` on FCOS, for me it's literally "podman". Moreover, on RHEL 7 with docker it sets the variable to `oci`, so testing for that would break the install logic. Just to compare, what is `rpm -qa|grep podman` for you? for me it is
```
podman-plugins-1.9.3-1.fc32.x86_64
podman-1.9.3-1.fc32.x86_64
```

**Comment by martinpitt** [comment]
Reopening with "needs more info" flag for the `$container` issue.

**Comment by wdouglascampbell** [comment]
I get the same results from ```rpm -qa|grep podman```.

I can reproduce the issue with $container returning oci by doing the following:

```
git clone https://github.com/cockpit-project/cockpit-container.git
cd container-build-area
podman build --tag localhost/cockpit/ws:latest .
podman container runlabel INSTALL localhost/cockpit/ws
```

which outputs

```
command: podman run --rm --privileged -v /:/host localhost/cockpit/ws:latest /container/atomic-install localhost/cockpit/ws:latest
+ sed -e /pam_selinux/d -e /pam_sepermit/d /etc/pam.d/cockpit
+ mkdir -p /host/etc/cockpit/ws-certs.d /host/etc/cockpit/machines.d
+ chmod 755 /host/etc/cockpit/ws-certs.d /host/etc/cockpit/machines.d
+ chown root:root /host/etc/cockpit/ws-certs.d /host/etc/cockpit/machines.d
+ mkdir -p /host/var/lib/cockpit
+ chmod 775 /host/var/lib/cockpit
+ chown root:wheel /host/var/lib/cockpit
+ mkdir -p /etc/ssh
+ '[' oci = podman ']'
+ /bin/mount --bind /host/etc/cockpit /etc/cockpit
+ /usr/sbin/remotectl certificate --ensure
```

I also get $container = oci if I just start a container based on the current Fedora 31 image.

**Comment by wdouglascampbell** [comment]
Sorry.  I forgot to indicate that since I can pull the image down now, it does run fine for me so it may not matter that when building the image from source that it doesn't work.

**Comment by fcrozat** [comment]
The issue is still present for people rebuilding cockpit on a different base container than Fedora, for instance on openSUSE (https://build.opensuse.org/package/show/devel:kubic:containers/opensuse-cockpit-ws-image ), unless we can the container environment variable.
